api/envoy/config/core/v3/address.proto

```diff
@@ -30,6 +30,18 @@ message Pipe {
   uint32 mode = 2 [(validate.rules).uint32 = {lte: 511}];
 }
 
+// [#not-implemented-hide:] The address representing an envoy internal listener, or a client connecting to that listener.
+message EnvoyInternalAddress {
+  oneof address_name_specifier {
+    option (validate.required) = true;
+
+    // The address is the name of a internal listener.
+    string server_listener_name = 1;
+
+    string client_address_id = 2;
```

Comment:
This field needs documenting, but in general, I think this entire message needs a lot more detail of when/how this is used and semantics. This should be standalone and not rely on the existing loopback issue.

Reply:
I don't have further plan on the `client_address_id`. In the prototype of the entire internal connection life time, I don't expect any reference to the content of `client_address_id`. 
`server_listener_name` has the constraint that server address must act as a reference as `listener_name` although this constraint cannot be enforced by annoation.

Meanwhile, `client_address` is not used anywhere but it's good to allow C++ internal address dumped to a protobuf message.
I can remove `client_address` for now since client address is not yet needed in future 3 PRs.

Reply:
`This should be standalone and not rely on the existing loopback issue.`
Do you mean I should add that this address is not loopback address?

Reply:
Removed client_address_id for now.
Also add the comment that server_listener_name will be used to look up message `Listener::listener_name`.

Reply:
Thanks, this is much clearer.